src/main/kotlin/com/wynntils/athena/database/DatabaseManager.kt

```diff
@@ -1,90 +1,59 @@
 package com.wynntils.athena.database
 
-import com.github.benmanes.caffeine.cache.Caffeine
-import com.rethinkdb.RethinkDB.r
-import com.rethinkdb.net.Result
+import com.mongodb.ConnectionString
+import com.mongodb.MongoClientSettings
+import com.mongodb.client.MongoDatabase
 import com.wynntils.athena.core.configs.databaseConfig
 import com.wynntils.athena.core.data.Location
-import com.wynntils.athena.database.objects.ApiKeyProfile
-import com.wynntils.athena.database.objects.GatheringSpotProfile
-import com.wynntils.athena.database.objects.GuildProfile
-import com.wynntils.athena.database.objects.UserProfile
+import com.wynntils.athena.database.objects.ApiKey
+import com.wynntils.athena.database.objects.GatheringSpot
+import com.wynntils.athena.database.objects.Guild
+import com.wynntils.athena.database.objects.User
+import org.bson.UuidRepresentation
+import org.litote.kmongo.*
 import java.util.*
 import java.util.concurrent.ExecutorService
 import java.util.concurrent.Executors
-import java.util.concurrent.TimeUnit
 
 object DatabaseManager {
```

Comment:
I know this is how I do a lotta stuff on the projects we've worked on but I'm personally looking at switching to dependency injection from singletons that contain data.

Take a look at https://insert-koin.io/, the main advantage is not wanting to die during unit tests (though I should note I haven't used it on a big project yet).

Reply:
That might be a good idea but this is not the commit to do that